### PR TITLE
Drone

### PR DESCRIPTION
**Pull request by martow**
Moved maximum velocity and acceleration from tactic check to engine. Created basic drone to drone communication.

### CODE REVIEW

File: implementation/common/src/main/java/org/inaetics/dronessimulator/common/protocol/MessageTopic.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
  * Different possible topics of messages for the Subscriber.
  */
 public enum MessageTopic implements Topic {
+    RADIO("RADIO"),
```
Comment (jjkester) [inline implementation/common/src/main/java/org/inaetics/dronessimulator/common/protocol/MessageTopic.java]: Can you document this topic as well?

File: implementation/common/src/main/java/org/inaetics/dronessimulator/common/protocol/MessageTopic.java [hunk 2]
```diff
@@ -19,6 +20,7 @@
      */
     ARCHITECTURE ("Architecture");
 
+
```
Comment (jjkester) [inline implementation/common/src/main/java/org/inaetics/dronessimulator/common/protocol/MessageTopic.java]: This blank line is not needed.

File: implementation/common/src/main/java/org/inaetics/dronessimulator/common/protocol/TextMessage.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package org.inaetics.dronessimulator.common.protocol;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TextMessage extends ProtocolMessage {
```
Comment (jjkester) [inline implementation/common/src/main/java/org/inaetics/dronessimulator/common/protocol/TextMessage.java]: Can you document the class?

File: implementation/common/src/main/java/org/inaetics/dronessimulator/common/protocol/TextMessage.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    /** Indentifier of object */
```
Comment (jjkester) [inline implementation/common/src/main/java/org/inaetics/dronessimulator/common/protocol/TextMessage.java]: This comment does not accurately describe the field.

File: implementation/drone/components/engine/src/java/org/inaetics/dronessimulator/drone/components/engine/Engine.java [hunk 5]
```diff
@@ -92,7 +130,9 @@ public void changeAcceleration(D3Vector input_acceleration){
         D3Vector acceleration = input_acceleration;
 
         acceleration = this.limit_acceleration(acceleration);
+        acceleration = this.limit_velocity(acceleration);
         acceleration = this.stagnate_acceleration(acceleration);
+
```
Comment (jjkester) [inline implementation/drone/components/engine/src/java/org/inaetics/dronessimulator/drone/components/engine/Engine.java]: This newline is not needed

File: implementation/drone/components/radar/src/java/org/inaetics/dronessimulator/drone/components/radar/Activator.java [hunk 6]
```diff
@@ -23,9 +24,14 @@ public void init(BundleContext bundleContext, DependencyManager dependencyManage
...
                 )
                 .add(createServiceDependency()
+                        .setService(Discoverer.class)
+                        .setRequired(true)
+                )
+                .add(createServiceDependency()
                     .setService(ArchitectureEventController.class)
                     .setRequired(true)
                 )
+
```
Comment (jjkester) [inline implementation/drone/components/radar/src/java/org/inaetics/dronessimulator/drone/components/radar/Activator.java]: This newline is not needed

File: implementation/drone/components/radio/src/java/org/inaetics/dronessimulator/drone/components/radio/Activator.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import org.apache.felix.dm.DependencyActivatorBase;
+import org.apache.felix.dm.DependencyManager;
+import org.inaetics.dronessimulator.drone.droneinit.DroneInit;
+import org.inaetics.dronessimulator.pubsub.api.publisher.Publisher;
+import org.inaetics.dronessimulator.pubsub.api.subscriber.Subscriber;
+import org.osgi.framework.BundleContext;
+
+/**
+ * Created by mart on 17-5-17.
+ */
```
Comment (jjkester) [inline implementation/drone/components/radio/src/java/org/inaetics/dronessimulator/drone/components/radio/Activator.java]: Actually document this

File: implementation/drone/components/radio/src/java/org/inaetics/dronessimulator/drone/components/radio/Radio.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import org.inaetics.dronessimulator.pubsub.api.Message;
+import org.inaetics.dronessimulator.pubsub.api.MessageHandler;
+import org.inaetics.dronessimulator.pubsub.api.subscriber.Subscriber;
+import org.inaetics.dronessimulator.pubsub.api.publisher.Publisher;
+
+
+import java.io.IOException;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+public class Radio implements MessageHandler {
```
Comment (jjkester) [inline implementation/drone/components/radio/src/java/org/inaetics/dronessimulator/drone/components/radio/Radio.java]: This whole class is not documented

File: implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java [hunk 9]
```diff
@@ -8,11 +8,15 @@
 import org.inaetics.dronessimulator.drone.components.gun.Gun;
 import org.inaetics.dronessimulator.drone.components.radar.Radar;
 
+
```
Comment (jjkester) [inline implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java]: This newline is not needed

File: implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java [hunk 10]
```diff
@@ -8,11 +8,15 @@
...
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.ThreadLocalRandom;
 
+/**
+ * Created by mart on 17-5-17.
+ */
```
Comment (jjkester) [inline implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java]: Actually document this class

File: implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java [hunk 11]
```diff
@@ -53,6 +57,12 @@ void calculateTactics(){
     /**
      *  -- FUNCTIONS
      */
+
+    /**
+     *
+     * @param input_acceleration
+     * @return
```
Comment (jjkester) [inline implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java]: Return value not documented, method not properly documented.

File: implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java [hunk 12]
```diff
@@ -65,20 +75,11 @@ private D3Vector accelerateByNoMovement(D3Vector input_acceleration){
...
-            double factor = 0.25;
-            output_acceleration = m_gps.getAcceleration().scale(factor);
-        }
-        return output_acceleration;
-    }
-
+    /**
+     *
+     * @param input_acceleration
+     * @return
```
Comment (jjkester) [inline implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java]: Return value not documented, method not properly documented.

File: implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java [hunk 13]
```diff
@@ -95,6 +96,11 @@ private D3Vector accelerateForWall(D3Vector input_acceleration){
         return output_acceleration;
     }
 
+    /**
+     *
+     * @param input_acceleration
+     * @return
+     */
```
Comment (jjkester) [inline implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java]: Method not properly documented.

File: implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java [hunk 14]
```diff
@@ -144,16 +150,20 @@ else if(m_gps.getPosition().getZ() <= 0){
         return output_acceleration;
     }
 
+    /**
+     *
+     */
```
Comment (jjkester) [inline implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java]: Method NOT documented

File: implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java [hunk 15]
```diff
@@ -144,16 +150,20 @@ else if(m_gps.getPosition().getZ() <= 0){
...
-        output_acceleration = this.accelerateByEngine(output_acceleration);
         output_acceleration = this.accelerateForWall(output_acceleration);
         output_acceleration = this.accelerateAfterWall(output_acceleration);
-        output_acceleration = m_engine.limit_acceleration(output_acceleration);
         m_engine.changeAcceleration(output_acceleration);
     }
 
+    /**
+     *
+     */
```
Comment (jjkester) [inline implementation/drone/tactic/src/java/org/inaetics/dronessimulator/drone/tactic/SimpleTactic.java]: Method not documented

**Review by jjkester — CHANGES_REQUESTED** [review]

**Comment by martow** [comment]
Fixed the documentation issues.

**Review by jjkester — APPROVED** [review]

**Pull request merged**